src/rgw/rgw_sal_motr.cc

```diff
@@ -452,8 +452,123 @@ int MotrBucket::remove_bucket(const DoutPrefixProvider *dpp, bool delete_childre
 {
   int ret;
 
+  ldpp_dout(dpp, 20) << "remove_bucket Entry=" << info.bucket.name << dendl;
+
+  // Refresh info
   ret = load_bucket(dpp, y);
-  // TODO: implement bucket removal
+  if (ret < 0) {
+    ldpp_dout(dpp, 0) << "ERROR: remove_bucket load_bucket failed rc=" << ret << dendl;
+    return ret;
+  }
+
+  ListParams params;
+  params.list_versions = true;
+  params.allow_unordered = true;
+
+  ListResults results;
+
+  // 1. Check if Bucket has objects.
+  // If bucket contains objects and delete_children is true, delete all objects.
+  // Else throw error that bucket is not empty.
+  do {
+    results.objs.clear();
+
+    // Check if bucket has objects.
+    ret = list(dpp, params, 1000, results, y);
+    if (ret < 0) {
+      return ret;
+    }
+
+    // If result contains entries, bucket is not empty.
+    if (!results.objs.empty() && !delete_children) {
+      ldpp_dout(dpp, 0) << "ERROR: could not remove non-empty bucket " << info.bucket.name << dendl;
+      return -ENOTEMPTY;
+    }
+
+    for (const auto& obj : results.objs) {
+      rgw_obj_key key(obj.key);
+      /* xxx dang */
+      ret = rgw_remove_object(dpp, store, this, key);
+      if (ret < 0 && ret != -ENOENT) {
+        ldpp_dout(dpp, 0) << "ERROR: remove_bucket rgw_remove_object failed rc=" << ret << dendl;
+	      return ret;
+      }
+    }
+  } while(results.is_truncated);
+
+  // 2. Abort Mp uploads on the bucket.
+  ret = abort_multiparts(dpp, store->ctx());
+  if (ret < 0) {
+    return ret;
+  }
+
+  // 3. Remove mp index??
+  string bucket_multipart_iname = "motr.rgw.bucket." + info.bucket.name + ".multiparts";
```

Comment:
please validate this scenario for expected behavior if bucket is empty and there are some ongoing multi parts, and delete_children is not true

Reply:
There is an abort_multipart function above this function call that will take care of this scenario, but its currently not implemented so cant test this scenario. JIRA for Abort Implementation (https://jts.seagate.com/browse/EOS-27630)